src/components/inputs/date/date.component.tsx

```diff
@@ -121,36 +123,30 @@ const DateField: React.FC<FormFieldProps> = ({ question, onChange, handler, prev
         <div className={styles.datetime}>
           <div>
             <Layer>
-              <DatePicker
-                datePickerType="single"
-                onChange={onDateChange}
-                className={styles.boldedLabel}
-                dateFormat={carbonDateFormat}>
-                <DatePickerInput
-                  id={question.id}
-                  placeholder={placeholder}
-                  labelText={
-                    question.isRequired ? (
-                      <RequiredFieldLabel label={t(question.label)} />
-                    ) : (
-                      <span>{t(question.label)}</span>
-                    )
-                  }
-                  value={field.value instanceof Date ? field.value.toLocaleDateString(locale) : field.value}
-                  // Added for testing purposes.
-                  // Notes:
-                  // Something strange is happening with the way events are propagated and handled by Carbon.
-                  // When we manually trigger an onchange event using the 'fireEvent' lib, the handler below will
-                  // be triggered as opposed to the former handler that only gets triggered at runtime.
-                  onChange={(e) => onDateChange([dayjs(e.target.value, placeholder.toUpperCase()).toDate()])}
-                  disabled={question.isDisabled}
-                  invalid={errors.length > 0}
-                  invalidText={errors[0]?.message}
-                  warn={warnings.length > 0}
-                  warnText={warnings[0]?.message}
-                  readOnly={question.readonly}
-                />
-              </DatePicker>
+              <OpenmrsDatePicker
+                id={question.id}
+                dateFormat={carbonDateFormat}
+                onChange={(date) => onDateChange([date])}
+                labelText={
+                  question.isRequired ? (
+                    <RequiredFieldLabel label={t(question.label)} />
+                  ) : (
+                    <span>{t(question.label)}</span>
+                  )
+                }
+                invalid={errors.length > 0}
+                invalidText={errors[0]?.message}
+                value={field.value}
+                carbonOptions={{
+                  placeholder: placeholder,
+                  warn: warnings[0]?.message,
+                  warnText: warnings[0]?.message,
+                  className: styles.boldedLabel,
+                  datePickerType: datePickerType,
+                }}
+                {...(typeof question.readonly === 'boolean' && { readonly: question.readonly })}
+                {...(typeof question.disabled === 'boolean' && { disabled: question.disabled })}
```

Comment:
Would this work?

```suggestion
                disabled={question.isDisabled}
                readonly={question. readonly}
```

Reply:
Not really
In RFE, the type of **question.readonly** is `string | boolean` and that of **question.disabled** is `boolean | Omit<DisableProps, "isDisabled">` whereas in core **readonly** and **disabled** are both of type `boolean.`
The solution was either to allow the `OpenmrsDatepicker props` in core accept the same exact types or handle them at FE level. 
I tried to do it in core but according to [this](https://github.com/openmrs/openmrs-esm-core/pull/1007#discussion_r1612280735) comment from @ibacher, this should be handled in FE and core only receives boolean types.

When I went through the code, I realized that for `readonly` for example, it accepts a string (expression) but after the expression is evaluated, the value is changed to a boolean. Given this background, it made sense to only pass these 2 if the data type is boolean so that the types in core stay intact.

Reply:
> In RFE, the type of question.readonly is string | boolean and that of question.disabled is boolean | Omit<DisableProps, "isDisabled"> whereas in core readonly and disabled are both of type boolean.

For the disabled property you're supposed to use `question.isDisabled` which is type boolean.

As for `readonly` and other related props we should update the typing to simply `boolean` because the [schema transformer parses the strings to booleans](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/transformers/default-schema-transformer.ts#L79).

Reply:
Just checked the schema `parseBooleanTokenIfPresent` function. 
From my understanding, it assumes that the the string is either "true" or "false"
However, we have cases where it is a totally different expression (I believe this wasn't a smart implementation) and I've seen it used in PMTCT forms.
Here is [how FE handles it](https://github.com/openmrs/openmrs-form-engine-lib/blob/main/src/components/encounter/encounter-form.component.tsx#L309C11-L322C12)
We need to agree if this is no longer desirable so that the forms where it is being used can also be adjusted accordingly.

Reply:
You make a good point but at this point it's still fair to assume that the `readonly` prop has already been processed to a boolean. As you can see in the snippet below the expression is stored in `field['readonlyExpression']`. (I think to be more consistent we should use the `readonly` vs `isReadonly` pattern)

```ts
if (typeof field.readonly == 'string' && field.readonly?.split(' ')?.length > 1) {
            // needed to store the expression for further evaluations
            field['readonlyExpression'] = field.readonly;
            field.readonly = evaluateExpression(

```

To make the TS compiler happy, use the `isTrue(string | boolean)` function:

```ts
readonly={isTrue(question.readonly)}
```

Reply:
This makes sense.

Done.